Docker backend build fails

I run into this error when building the backend docker container after running `docker compose up`:

```
#24 [build-prod 5/5] RUN SQLX_OFFLINE=true cargo build --release
#24 1.688    Compiling arcadia-backend v0.1.0 (/app)
#24 4.107 error[E0658]: `let` expressions in this position are unstable
#24 4.107    --> src/repositories/auth_repository.rs:175:28
#24 4.107     |
#24 4.107 175 |                         if let Some(code) = code
#24 4.107     |                            ^^^^^^^^^^^^^^^^^^^^^
#24 4.107     |
#24 4.107     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
#24 4.107 
#24 12.35 For more information about this error, try `rustc --explain E0658`.
#24 12.38 error: could not compile `arcadia-backend` (lib) due to 1 previous error
#24 ERROR: process "/bin/sh -c SQLX_OFFLINE=true cargo build --release" did not complete successfully: exit code: 101
------
 > [build-prod 5/5] RUN SQLX_OFFLINE=true cargo build --release:
4.107 error[E0658]: `let` expressions in this position are unstable
4.107    --> src/repositories/auth_repository.rs:175:28
4.107     |
4.107 175 |                         if let Some(code) = code
4.107     |                            ^^^^^^^^^^^^^^^^^^^^^
4.107     |
4.107     = note: see issue #53667 <https://github.com/rust-lang/rust/issues/53667> for more information
4.107 
12.35 For more information about this error, try `rustc --explain E0658`.
12.38 error: could not compile `arcadia-backend` (lib) due to 1 previous error
------
Dockerfile:43

--------------------

  41 |     COPY . .

  42 |     

  43 | >>> RUN SQLX_OFFLINE=true cargo build --release

  44 |     

  45 |     FROM prebuild AS build-debug

--------------------

failed to solve: process "/bin/sh -c SQLX_OFFLINE=true cargo build --release" did not complete successfully: exit code: 101
```

--- a/backend/Dockerfile
+++ b/backend/Dockerfile
@@ -4,7 +4,7 @@ RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloa
 
 FROM alpine:3 AS alpine-base
 
-FROM rust:1.86-slim-bookworm AS prebuild
+FROM rust:1.89-slim-bookworm AS prebuild
 # Don't delete any downloaded packages, we'd be using BuildKit cache mounts later!
 RUN rm -f /etc/apt/apt.conf.d/docker-clean; echo 'Binary::apt::APT::Keep-Downloaded-Packages "true";' > /etc/apt/apt.conf.d/keep-cache
 

